frontend/src/app/organization/organization-contact/organization-contact.component.ts

```diff
@@ -1,46 +1,67 @@
-import { Component, EventEmitter, Input, OnChanges, Output, SimpleChanges, ViewChild } from '@angular/core';
-import { NgForm } from '@angular/forms';
+import { Component, OnInit, OnChanges, SimpleChanges, EventEmitter, Output, Input } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
 
 import { OrganizationContactPayload } from '../organization.model';
 import { ContactStatus } from '../organization.interfaces';
-import {AuthService} from '../../auth.service';
-import {Observable} from 'rxjs/Observable';
-import {User} from '../../user';
-import {UserService} from '../../user.service';
-
+import { AuthService } from '../../auth.service';
+import { Observable } from 'rxjs/Observable';
+import { User } from '../../user';
+import { UserService } from '../../user.service';
+import { OrganizationService } from '../organization.service';
 
 @Component({
   selector: 'volontulo-organization-contact',
   templateUrl: './organization-contact.component.html',
   styleUrls: ['./organization-contact.component.scss']
 })
-export class OrganizationContactComponent implements OnChanges {
-  @ViewChild('contactForm') contactForm: NgForm;
-  @Output() contact = new EventEmitter<OrganizationContactPayload>();
+
+export class OrganizationContactComponent implements OnInit, OnChanges {
   @Input() contactStatus: ContactStatus;
+  @Output() contact = new EventEmitter<OrganizationContactPayload>();
   submitDisabled = false;
-  alertSuccessClosed = true;
-  alertErrorClosed = true;
   user$: Observable<User> = this.authService.user$;
   getFullName = this.userService.getFullName;
 
-  constructor(private authService: AuthService, private userService: UserService) {
+  public fg: FormGroup = this.fb.group({
+    name: ['', [Validators.required, Validators.minLength(3), Validators.maxLength(30)]],
+    email: ['', [Validators.required, Validators.minLength(3), Validators.maxLength(30), Validators.email]],
+    phone_no: ['', [Validators.required, Validators.minLength(9), Validators.maxLength(9), Validators.pattern(/^[0-9]{9}$/)]],
+    message: ['', [Validators.required, Validators.minLength(10), Validators.maxLength(2000)]],
+    honeyBunny: ['']
+  });
+  public success: null | boolean = null;
+
+
+  constructor(
+    private fb: FormBuilder,
+    private authService: AuthService,
+    private userService: UserService,
+    private organizationService: OrganizationService
+  ) { }
+
+  ngOnInit() {
+    this.user$.subscribe(user => {
+      this.fg.controls.name.setValue(this.getFullName(user));
+      this.fg.controls.email.setValue(user.email);
+      this.fg.controls.phone_no.setValue(user.phoneNo);
+    });
```

Comment:
I found bug there - i don't check whether is user login, or not (if not that will throw error to the console). I will fix it with an approach like in contact.component.ts